src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/HashProviderCng.cs

```diff
@@ -39,7 +39,7 @@ internal HashProviderCng(string hashAlgId, ReadOnlySpan<byte> key, bool isHmac)
             // So keep hHash trapped in this scope to prevent (mis-)use of it.
             {
                 SafeBCryptHashHandle hHash;
-                NTSTATUS ntStatus = Interop.BCrypt.BCryptCreateHash(_hAlgorithm, out hHash, IntPtr.Zero, 0, key, key == null ? 0 : key.Length, BCryptCreateHashFlags.BCRYPT_HASH_REUSABLE_FLAG);
+                NTSTATUS ntStatus = Interop.BCrypt.BCryptCreateHash(_hAlgorithm, out hHash, IntPtr.Zero, 0, key, key.Length, BCryptCreateHashFlags.BCRYPT_HASH_REUSABLE_FLAG);
```

Comment:
Be careful here. This _might_ be ok, but there are some places in crypto where @bartonjs has made a distinction between a 0-length span and a span wrapping a null pointer. I don't know if this is one of them.  Ideally we'd have comments next to all of the cases where that behavior is being used (if we don't already), or maybe we instead do the comparison very explicitly, using Unsafe.IsNullRef on the MemoryMarshal.GetReference from the span. @bartonjs?

Reply:
This one is okay. It got spanified in https://github.com/dotnet/runtime/commit/32388df1ad99d616377b5ec6ca4966d60f10b365

but the `null` handling of the array was not updated.